docs: 960 docs add a glossary concept section (#970)
argilla-io/distilabel: docs/sections/how_to_guides/basic/concepts.md

# Concepts

## Pipeline

Distilabel provides all the tools you need to your scalable and reliable pipelines for synthetic data generation and AI-feedback. Pipelines are used to generate data, evaluate models, manipulate data, or any other general task. They are made up of different components: Steps, Tasks and LLMs, which are chained together in a directed acyclic graph (DAG).

> - Steps are the building blocks of your pipeline. Normal steps are used for basic executions like loading data, applying some transformations, or any other general task.

Reviewer: ```suggestion
- **Steps**: These are the building blocks of your pipeline. Normal steps are used for basic executions like loading data, applying some transformations, or any other general task.
```

> - Tasks are steps that rely on LLMs and prompts to perform more specific tasks. For example, they can be used to generate data, evaluate models or manipulate data.

Reviewer: ```suggestion
- **Tasks**: These are steps that rely on LLMs and prompts to perform generative tasks. For example, they can be used to generate data, evaluate models or manipulate data.
```

> - LLMs are the models that will perform the task. They can be local or remote models, and open-source or commercial models.

Reviewer: ```suggestion
- **LLMs**: These are the models that will perform the task. They can be local or remote models, and open-source or commercial models.
```

![Pipeline](../../../assets/pipelines/generate-preference-dataset.png)

### Minimal code example

`distilabel` gives a lot of flexibility to create your pipelines, but to start right away, you can omit a lot of the details and let default values:

```python
from distilabel.llms import InferenceEndpointsLLM
from distilabel.pipeline import Pipeline
from distilabel.steps.tasks import TextGeneration
from distilabel.steps import LoadDataFromDict
from datasets import load_dataset

dataset = load_dataset("distilabel-internal-testing/instruction-dataset-mini", split="test")

with Pipeline() as pipeline:  # (1)
    load_data_from_dict = LoadDataFromDict(
        data={
            "instruction": dataset["instruction"],
            "response": dataset["response"],
        }
    ) # (2)
    text_generation = TextGeneration(
        llm=InferenceEndpointsLLM(
            model_id="meta-llama/Meta-Llama-3.1-8B-Instruct"
        ),
        prompt=PromptTemplate(
            template="Generate a short text: {instruction}"
        ),
    )  # (3)

>     load_data_from_dict.connect(text_generation)

Reviewer: I would add a comment to this line, something like: "Connects steps in the pipeline: `text_generation` will run after `load_data_from_dict`."

if __name__ == "__main__":
    distiset = pipeline.run()

>     distiset.push_to_hub(repo_id="distilabel-example")

Reviewer: I would also add a comment here: "Optionally, you can push the final result to the Huggingface Hub"

```

> 1. The [`Pipeline`][distilabel.pipeline.Pipeline].

Reviewer: ```suggestion
1. The [`Pipeline`][distilabel.pipeline.Pipeline] gathers all steps and tasks to be executed.
```

> 2. The [`Step`][distilabel.steps.base.Step] that loads data from a dictionary.

Reviewer: ```suggestion
2. This [`Step`][distilabel.steps.base.Step] loads data from a dictionary.
```
```suggestion
2. The [`Step`][distilabel.steps.base.Step] that loads data from a dictionary.
```

> 3. The [`Task`][distilabel.steps.tasks.base.Task] that executes a prompt.

Reviewer: ```suggestion
3. This [`Task`][distilabel.steps.tasks.base.Task] uses the instructions from the loaded dataset and the provided template to prompt an LLM.
```
